[Operator] UpdateTest.testImageChange throws TimeoutException

### Before reporting an issue

- [x] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

operator

### Describe the bug

```
Error:  Tests run: 11, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 675.086 s <<< FAILURE! - in org.keycloak.operator.testsuite.integration.UpdateTest
Error:  org.keycloak.operator.testsuite.integration.UpdateTest.testImageChange(UpdateStrategy)[3]  Time elapsed: 140.421 s  <<< FAILURE!
java.lang.AssertionError: java.util.concurrent.TimeoutException
	at org.keycloak.operator.testsuite.integration.UpdateTest.await(UpdateTest.java:281)
	at org.keycloak.operator.testsuite.integration.UpdateTest.testImageChange(UpdateTest.java:71)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at io.quarkus.test.junit.QuarkusTestExtension.runExtensionMethod(QuarkusTestExtension.java:960)
	at io.quarkus.test.junit.QuarkusTestExtension.runExtensionMethod(QuarkusTestExtension.java:890)
	at io.quarkus.test.junit.QuarkusTestExtension.interceptTestTemplateMethod(QuarkusTestExtension.java:850)
	at java.base/java.util.Optional.ifPresent(Optional.java:178)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
	at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1939)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
	at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:276)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1708)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
	at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:276)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1708)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
Caused by: java.util.concurrent.TimeoutException
	at java.base/java.util.concurrent.CompletableFuture.timedGet(CompletableFuture.java:1960)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2095)
	at org.keycloak.operator.testsuite.integration.UpdateTest.await(UpdateTest.java:279)
	... 40 more
```

## Runs:
- https://github.com/keycloak/keycloak/actions/runs/16051844660/job/45296507608?pr=40762
- https://github.com/keycloak/keycloak/actions/runs/16028590794/job/45223307463?pr=40867


More logs:
```
2025-07-03 14:02:29,022 WARN  [io.jav.ope.pro.eve.EventProcessor] (ReconcilerExecutor-keycloakcontroller-68) Resource Kubernetes Resource Creator/Update Conflict during reconciliation. Message: Failure executing: PATCH at: [https://10.96.0.1:443/apis/k8s.keycloak.org/v2alpha1/namespaces/keycloak-test-51e4ea3e-f396-4d43-bce3-810b764f91c9/keycloaks/example-kc/status?fieldManager=keycloakcontroller&force=true.](https://10.96.0.1/apis/k8s.keycloak.org/v2alpha1/namespaces/keycloak-test-51e4ea3e-f396-4d43-bce3-810b764f91c9/keycloaks/example-kc/status?fieldManager=keycloakcontroller&force=true.) Message: Operation cannot be fulfilled on keycloaks.k8s.keycloak.org "example-kc": the object has been modified; please apply your changes to the latest version and try again. Received status: Status(apiVersion=v1, code=409, details=StatusDetails(causes=[], group=k8s.keycloak.org, kind=keycloaks, name=example-kc, retryAfterSeconds=null, uid=null, additionalProperties={}), kind=Status, message=Operation cannot be fulfilled on keycloaks.k8s.keycloak.org "example-kc": the object has been modified; please apply your changes to the latest version and try again, metadata=ListMeta(_continue=null, remainingItemCount=null, resourceVersion=null, selfLink=null, additionalProperties={}), reason=Conflict, status=Failure, additionalProperties={}). Resource name: null
2025-07-03 14:02:29,096 WARN  [io.jav.ope.pro.eve.EventProcessor] (ReconcilerExecutor-keycloakcontroller-75) Resource Kubernetes Resource Creator/Update Conflict during reconciliation. Message: Failure executing: PATCH at: [https://10.96.0.1:443/apis/k8s.keycloak.org/v2alpha1/namespaces/keycloak-test-51e4ea3e-f396-4d43-bce3-810b764f91c9/keycloaks/example-kc/status?fieldManager=keycloakcontroller&force=true.](https://10.96.0.1/apis/k8s.keycloak.org/v2alpha1/namespaces/keycloak-test-51e4ea3e-f396-4d43-bce3-810b764f91c9/keycloaks/example-kc/status?fieldManager=keycloakcontroller&force=true.) Message: Operation cannot be fulfilled on keycloaks.k8s.keycloak.org "example-kc": the object has been modified; please apply your changes to the latest version and try again. Received status: Status(apiVersion=v1, code=409, details=StatusDetails(causes=[], group=k8s.keycloak.org, kind=keycloaks, name=example-kc, retryAfterSeconds=null, uid=null, additionalProperties={}), kind=Status, message=Operation cannot be fulfilled on keycloaks.k8s.keycloak.org "example-kc": the object has been modified; please apply your changes to the latest version and try again, metadata=ListMeta(_continue=null, remainingItemCount=null, resourceVersion=null, selfLink=null, additionalProperties={}), reason=Conflict, status=Failure, additionalProperties={}). Resource name: null
2025-07-03 14:02:40,464 WARN  [io.jav.ope.pro.eve.EventProcessor] (ReconcilerExecutor-keycloakcontroller-116) Resource Kubernetes Resource Creator/Update Conflict during reconciliation. Message: Failure executing: PATCH at: [https://10.96.0.1:443/apis/k8s.keycloak.org/v2alpha1/namespaces/keycloak-test-51e4ea3e-f396-4d43-bce3-810b764f91c9/keycloaks/example-kc/status?fieldManager=keycloakcontroller&force=true.](https://10.96.0.1/apis/k8s.keycloak.org/v2alpha1/namespaces/keycloak-test-51e4ea3e-f396-4d43-bce3-810b764f91c9/keycloaks/example-kc/status?fieldManager=keycloakcontroller&force=true.) Message: Operation cannot be fulfilled on keycloaks.k8s.keycloak.org "example-kc": the object has been modified; please apply your changes to the latest version and try again. Received status: Status(apiVersion=v1, code=409, details=StatusDetails(causes=[], group=k8s.keycloak.org, kind=keycloaks, name=example-kc, retryAfterSeconds=null, uid=null, additionalProperties={}), kind=Status, message=Operation cannot be fulfilled on keycloaks.k8s.keycloak.org "example-kc": the object has been modified; please apply your changes to the latest version and try again, metadata=ListMeta(_continue=null, remainingItemCount=null, resourceVersion=null, selfLink=null, additionalProperties={}), reason=Conflict, status=Failure, additionalProperties={}). Resource name: null
2025-07-03 14:03:50,746 WARN  [org.key.ope.upd.imp.AutoUpdateLogic] (ReconcilerExecutor-keycloakcontroller-122) Container has an unexpected error for Update Job
2025-07-03 14:03:50,914 WARN  [org.key.ope.upd.imp.AutoUpdateLogic] (ReconcilerExecutor-keycloakcontroller-121) Unexpected Update Job exit code: 137
2025-07-03 14:03:51,072 WARN  [org.key.ope.upd.imp.AutoUpdateLogic] (ReconcilerExecutor-keycloakcontroller-121) Unexpected Update Job exit code: 137
2025-07-03 14:03:51,467 WARN  [io.jav.ope.pro.eve.EventProcessor] (ReconcilerExecutor-keycloakcontroller-124) Resource Kubernetes Resource Creator/Update Conflict during reconciliation. Message: Failure executing: PATCH at: [https://10.96.0.1:443/apis/k8s.keycloak.org/v2alpha1/namespaces/keycloak-test-51e4ea3e-f396-4d43-bce3-810b764f91c9/keycloaks/example-kc/status?fieldManager=keycloakcontroller&force=true.](https://10.96.0.1/apis/k8s.keycloak.org/v2alpha1/namespaces/keycloak-test-51e4ea3e-f396-4d43-bce3-810b764f91c9/keycloaks/example-kc/status?fieldManager=keycloakcontroller&force=true.) Message: Operation cannot be fulfilled on keycloaks.k8s.keycloak.org "example-kc": the object has been modified; please apply your changes to the latest version and try again. Received status: Status(apiVersion=v1, code=409, details=StatusDetails(causes=[], group=k8s.keycloak.org, kind=keycloaks, name=example-kc, retryAfterSeconds=null, uid=null, additionalProperties={}), kind=Status, message=Operation cannot be fulfilled on keycloaks.k8s.keycloak.org "example-kc": the object has been modified; please apply your changes to the latest version and try again, metadata=ListMeta(_continue=null, remainingItemCount=null, resourceVersion=null, selfLink=null, additionalProperties={}), reason=Conflict, status=Failure, additionalProperties={}). Resource name: null
2025-07-03 14:03:51,579 ERROR [io.jav.ope.pro.eve.EventProcessor] (ReconcilerExecutor-keycloakcontroller-121) Error during event processing ExecutionScope{ resource id: ResourceID{name='example-kc', namespace='keycloak-test-51e4ea3e-f396-4d43-bce3-810b764f91c9'}, version: 6997}: io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: PATCH at: [https://10.96.0.1:443/apis/k8s.keycloak.org/v2alpha1/namespaces/keycloak-test-51e4ea3e-f396-4d43-bce3-810b764f91c9/keycloaks/example-kc/status?fieldManager=keycloakcontroller&force=true.](https://10.96.0.1/apis/k8s.keycloak.org/v2alpha1/namespaces/keycloak-test-51e4ea3e-f396-4d43-bce3-810b764f91c9/keycloaks/example-kc/status?fieldManager=keycloakcontroller&force=true.) Message: Operation cannot be fulfilled on keycloaks.k8s.keycloak.org "example-kc": the object has been modified; please apply your changes to the latest version and try again. Received status: Status(apiVersion=v1, code=409, details=StatusDetails(causes=[], group=k8s.keycloak.org, kind=keycloaks, name=example-kc, retryAfterSeconds=null, uid=null, additionalProperties={}), kind=Status, message=Operation cannot be fulfilled on keycloaks.k8s.keycloak.org "example-kc": the object has been modified; please apply your changes to the latest version and try again, metadata=ListMeta(_continue=null, remainingItemCount=null, resourceVersion=null, selfLink=null, additionalProperties={}), reason=Conflict, status=Failure, additionalProperties={}).
	at io.fabric8.kubernetes.client.KubernetesClientException.copyAsCause(KubernetesClientException.java:205)
	at io.fabric8.kubernetes.client.dsl.internal.OperationSupport.waitForResult(OperationSupport.java:507)
	at io.fabric8.kubernetes.client.dsl.internal.OperationSupport.handleResponse(OperationSupport.java:524)
	at io.fabric8.kubernetes.client.dsl.internal.OperationSupport.handlePatch(OperationSupport.java:419)
	at io.fabric8.kubernetes.client.dsl.internal.OperationSupport.handlePatch(OperationSupport.java:397)
	at io.fabric8.kubernetes.client.dsl.internal.BaseOperation.handlePatch(BaseOperation.java:764)
	at io.fabric8.kubernetes.client.dsl.internal.HasMetadataOperation.lambda$patch$2(HasMetadataOperation.java:231)
	at io.fabric8.kubernetes.client.dsl.internal.HasMetadataOperation.patch(HasMetadataOperation.java:236)
	at io.fabric8.kubernetes.client.dsl.internal.HasMetadataOperation.patch(HasMetadataOperation.java:251)
	at io.fabric8.kubernetes.client.dsl.internal.HasMetadataOperation.patch(HasMetadataOperation.java:44)
	at io.javaoperatorsdk.operator.processing.event.ReconciliationDispatcher$CustomResourceFacade.patchStatus(ReconciliationDispatcher.java:445)
	at io.javaoperatorsdk.operator.processing.event.ReconciliationDispatcher.handleErrorStatusHandler(ReconciliationDispatcher.java:216)
	at io.javaoperatorsdk.operator.processing.event.ReconciliationDispatcher.handleReconcile(ReconciliationDispatcher.java:132)
	at io.javaoperatorsdk.operator.processing.event.ReconciliationDispatcher.handleDispatch(ReconciliationDispatcher.java:97)
	at io.javaoperatorsdk.operator.processing.event.ReconciliationDispatcher.handleExecution(ReconciliationDispatcher.java:68)
	at io.javaoperatorsdk.operator.processing.event.EventProcessor$ReconcilerExecutor.run(EventProcessor.java:467)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: PATCH at: [https://10.96.0.1:443/apis/k8s.keycloak.org/v2alpha1/namespaces/keycloak-test-51e4ea3e-f396-4d43-bce3-810b764f91c9/keycloaks/example-kc/status?fieldManager=keycloakcontroller&force=true.](https://10.96.0.1/apis/k8s.keycloak.org/v2alpha1/namespaces/keycloak-test-51e4ea3e-f396-4d43-bce3-810b764f91c9/keycloaks/example-kc/status?fieldManager=keycloakcontroller&force=true.) Message: Operation cannot be fulfilled on keycloaks.k8s.keycloak.org "example-kc": the object has been modified; please apply your changes to the latest version and try again. Received status: Status(apiVersion=v1, code=409, details=StatusDetails(causes=[], group=k8s.keycloak.org, kind=keycloaks, name=example-kc, retryAfterSeconds=null, uid=null, additionalProperties={}), kind=Status, message=Operation cannot be fulfilled on keycloaks.k8s.keycloak.org "example-kc": the object has been modified; please apply your changes to the latest version and try again, metadata=ListMeta(_continue=null, remainingItemCount=null, resourceVersion=null, selfLink=null, additionalProperties={}), reason=Conflict, status=Failure, additionalProperties={}).
	at io.fabric8.kubernetes.client.dsl.internal.OperationSupport.requestFailure(OperationSupport.java:642)
	at io.fabric8.kubernetes.client.dsl.internal.OperationSupport.requestFailure(OperationSupport.java:622)
	at io.fabric8.kubernetes.client.dsl.internal.OperationSupport.assertResponseCode(OperationSupport.java:582)
	at io.fabric8.kubernetes.client.dsl.internal.OperationSupport.lambda$handleResponse$0(OperationSupport.java:549)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:646)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2179)
	at io.fabric8.kubernetes.client.http.StandardHttpClient.lambda$completeOrCancel$10(StandardHttpClient.java:141)
	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863)
	at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:841)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2179)
	at io.fabric8.kubernetes.client.http.ByteArrayBodyHandler.onBodyDone(ByteArrayBodyHandler.java:51)
	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863)
	at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:841)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2179)
	at io.fabric8.kubernetes.client.vertx.VertxHttpRequest.lambda$consumeBytes$1(VertxHttpRequest.java:120)
	at io.vertx.core.impl.ContextInternal.dispatch(ContextInternal.java:270)
	at io.vertx.core.impl.ContextInternal.dispatch(ContextInternal.java:252)
	at io.vertx.core.http.impl.HttpEventHandler.handleEnd(HttpEventHandler.java:76)
	at io.vertx.core.http.impl.HttpClientResponseImpl.handleEnd(HttpClientResponseImpl.java:250)
	at io.vertx.core.http.impl.Http1xClientConnection$StreamImpl.lambda$new$0(Http1xClientConnection.java:424)
	at io.vertx.core.streams.impl.InboundBuffer.handleEvent(InboundBuffer.java:279)
	at io.vertx.core.streams.impl.InboundBuffer.write(InboundBuffer.java:157)
	at io.vertx.core.http.impl.Http1xClientConnection$StreamImpl.handleEnd(Http1xClientConnection.java:734)
	at io.vertx.core.impl.ContextImpl.lambda$execute$7(ContextImpl.java:329)
	at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:173)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:166)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:569)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:998)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	... 1 more

2025-07-03 14:04:02,726 WARN  [io.jav.ope.pro.eve.EventProcessor] (ReconcilerExecutor-keycloakcontroller-129) Resource Kubernetes Resource Creator/Update Conflict during reconciliation. Message: Failure executing: PATCH at: [https://10.96.0.1:443/apis/k8s.keycloak.org/v2alpha1/namespaces/keycloak-test-51e4ea3e-f396-4d43-bce3-810b764f91c9/keycloaks/example-kc/status?fieldManager=keycloakcontroller&force=true.](https://10.96.0.1/apis/k8s.keycloak.org/v2alpha1/namespaces/keycloak-test-51e4ea3e-f396-4d43-bce3-810b764f91c9/keycloaks/example-kc/status?fieldManager=keycloakcontroller&force=true.) Message: Operation cannot be fulfilled on keycloaks.k8s.keycloak.org "example-kc": the object has been modified; please apply your changes to the latest version and try again. Received status: Status(apiVersion=v1, code=409, details=StatusDetails(causes=[], group=k8s.keycloak.org, kind=keycloaks, name=example-kc, retryAfterSeconds=null, uid=null, additionalProperties={}), kind=Status, message=Operation cannot be fulfilled on keycloaks.k8s.keycloak.org "example-kc": the object has been modified; please apply your changes to the latest version and try again, metadata=ListMeta(_continue=null, remainingItemCount=null, resourceVersion=null, selfLink=null, additionalProperties={}), reason=Conflict, status=Failure, additionalProperties={}). Resource name: null

2025-07-03 14:06:11,522 INFO  [org.key.ope.tes.int.BaseOperatorTest] (main) Deleting Keycloak CR
2025-07-03 14:06:11,582 INFO  [org.key.ope.tes.int.BaseOperatorTest] (main) 
------- STARTING: UpdateTest.testOptimizedImage-RECREATE_ON_IMAGE_CHANGE
------- Namespace: keycloak-test-51e4ea3e-f396-4d43-bce3-810b764f91c9
------- Mode: remote
2025-07-03 14:06:11,680 INFO  [org.key.ope.tes.uti.K8sUtils] (main) Waiting for Keycloak "example-kc"
2025-07-03 14:06:51,974 INFO  [org.key.ope.tes.uti.K8sUtils] (main) Waiting for Keycloak "example-kc"
2025-07-03 14:07:13,106 INFO  [org.key.ope.tes.int.BaseOperatorTest] (main) Deleting Keycloak CR
2025-07-03 14:07:13,173 INFO  [org.key.ope.tes.int.BaseOperatorTest] (main) 
------- STARTING: UpdateTest.testOptimizedImage-AUTO
------- Namespace: keycloak-test-51e4ea3e-f396-4d43-bce3-810b764f91c9
------- Mode: remote
2025-07-03 14:07:13,264 INFO  [org.key.ope.tes.uti.K8sUtils] (main) Waiting for Keycloak "example-kc"
2025-07-03 14:07:53,546 INFO  [org.key.ope.tes.uti.K8sUtils] (main) Waiting for Keycloak "example-kc"
2025-07-03 14:08:12,526 INFO  [org.key.ope.tes.int.BaseOperatorTest] (main) Deleting Keycloak CR
2025-07-03 14:08:12,694 INFO  [org.key.ope.tes.int.BaseOperatorTest] (main) 
------- STARTING: UpdateTest.testOptimizedImage-EXPLICIT
------- Namespace: keycloak-test-51e4ea3e-f396-4d43-bce3-810b764f91c9
------- Mode: remote
2025-07-03 14:08:12,758 INFO  [org.key.ope.tes.uti.K8sUtils] (main) Waiting for Keycloak "example-kc"
2025-07-03 14:08:53,052 INFO  [org.key.ope.tes.uti.K8sUtils] (main) Waiting for Keycloak "example-kc"
2025-07-03 14:08:54,057 INFO  [org.key.ope.tes.int.BaseOperatorTest] (main) Deleting Keycloak CR
2025-07-03 14:08:54,146 INFO  [org.key.ope.tes.int.BaseOperatorTest] (main) Deleting RBAC from Namespace keycloak-test-51e4ea3e-f396-4d43-bce3-810b764f91c9
2025-07-03 14:08:54,293 INFO  [org.key.ope.tes.int.BaseOperatorTest] (main) Deleting namespace : keycloak-test-51e4ea3e-f396-4d43-bce3-810b764f91c9
```

### Version

main

### Regression

- [ ] The issue is a regression

### Expected behavior

test pass

### Actual behavior

test failed

### How to Reproduce?

See the CI runs

### Anything else?

_No response_

--- a/operator/src/main/java/org/keycloak/operator/controllers/KeycloakController.java
+++ b/operator/src/main/java/org/keycloak/operator/controllers/KeycloakController.java
@@ -127,7 +127,7 @@ public class KeycloakController implements Reconciler<Keycloak> {
             return UpdateControl.patchResource(patchedKc);
         }
 
-        var existingDeployment = context.getSecondaryResource(StatefulSet.class).orElse(null);
+        var existingDeployment = context.getSecondaryResource(StatefulSet.class).filter(ss -> ss.hasOwnerReferenceFor(kc)).orElse(null);
         ContextUtils.storeOperatorConfig(context, config);
         ContextUtils.storeWatchedResources(context, watchedResources);
         ContextUtils.storeDistConfigurator(context, distConfigurator);

--- a/operator/src/main/java/org/keycloak/operator/controllers/KeycloakRealmImportController.java
+++ b/operator/src/main/java/org/keycloak/operator/controllers/KeycloakRealmImportController.java
@@ -58,7 +58,7 @@ public class KeycloakRealmImportController implements Reconciler<KeycloakRealmIm
 
         var statusBuilder = new KeycloakRealmImportStatusBuilder();
 
-        Job existingJob = context.getSecondaryResource(Job.class).orElse(null);
+        Job existingJob = context.getSecondaryResource(Job.class).filter(job -> job.hasOwnerReferenceFor(realm)).orElse(null);
         StatefulSet existingDeployment = context.getClient().resources(StatefulSet.class).inNamespace(realm.getMetadata().getNamespace())
                 .withName(realm.getSpec().getKeycloakCRName()).get();
 

--- a/operator/src/main/java/org/keycloak/operator/controllers/KeycloakUpdateJobDependentResource.java
+++ b/operator/src/main/java/org/keycloak/operator/controllers/KeycloakUpdateJobDependentResource.java
@@ -102,7 +102,7 @@ public class KeycloakUpdateJobDependentResource extends CRUDKubernetesDependentR
     public static boolean isJobFromCurrentKeycloakCr(Job job, Keycloak keycloak) {
         var annotations = job.getMetadata().getAnnotations();
         var hash = annotations.get(KEYCLOAK_CR_HASH_ANNOTATION);
-        return Objects.equals(hash, keycloakHash(keycloak));
+        return job.hasOwnerReferenceFor(keycloak) && Objects.equals(hash, keycloakHash(keycloak));
     }
 
     public static String jobName(Keycloak keycloak) {

--- a/operator/src/main/java/org/keycloak/operator/update/impl/AutoUpdateLogic.java
+++ b/operator/src/main/java/org/keycloak/operator/update/impl/AutoUpdateLogic.java
@@ -54,14 +54,14 @@ public class AutoUpdateLogic extends BaseUpdateLogic {
             return Optional.of(UpdateControl.noUpdate());
         }
 
-        if (isJobRunning(existingJob.get())) {
-            Log.debug("Update Job is running. Waiting until terminated.");
-            return Optional.of(UpdateControl.noUpdate());
-        }
-
         // Keycloak CR may be updated while the job is running; we need to delete and start over.
         if (!KeycloakUpdateJobDependentResource.isJobFromCurrentKeycloakCr(existingJob.get(), keycloak)) {
             context.getClient().resource(existingJob.get()).lockResourceVersion().delete();
+            return Optional.of(UpdateControl.noUpdate());
+        }
+
+        if (isJobRunning(existingJob.get())) {
+            Log.debug("Update Job is running. Waiting until terminated.");
             return Optional.of(UpdateControl.noUpdate());
         }
 

